db_store/src/iam_auth_pool.rs

```diff
@@ -0,0 +1,187 @@
+        .and_then(|creds| {
+            Ok(Credentials::new(
+                creds.access_key_id().ok_or_else(|| {
+                    Error::InvalidAssumedCredentials("no access_key_id".to_string())
+                })?,
+                creds.secret_access_key().ok_or_else(|| {
+                    Error::InvalidAssumedCredentials("no secret_access_key".to_string())
+                })?,
+                creds.session_token().map(|s| s.to_string()),
+                creds
+                    .expiration()
+                    .map(|e| UNIX_EPOCH + Duration::from_secs(e.secs() as u64)),
+                "Helium Foundation",
+            ))
+        })
+}
+
+fn region(settings: &Settings) -> Result<Region> {
+    settings
+        .iam_region
+        .as_ref()
+        .map(|r| Region::new(r.clone()))
+        .ok_or_else(|| Error::InvalidConfiguration("iam_region is required".to_string()))
+}
+
+async fn assume_role(
+    client: &aws_sdk_sts::Client,
+    settings: &Settings,
+) -> Result<aws_sdk_sts::output::AssumeRoleOutput> {
+    client
+        .assume_role()
+        .role_arn(
+            settings.iam_role_arn.as_ref().ok_or_else(|| {
+                Error::InvalidConfiguration("iam_role_arn is required".to_string())
+            })?,
+        )
+        .role_session_name(settings.iam_role_session_name.as_ref().ok_or_else(|| {
+            Error::InvalidConfiguration("iam_role_session_name is required".to_string())
+        })?)
+        .duration_seconds(settings.iam_duration_seconds.ok_or_else(|| {
+            Error::InvalidConfiguration("iam_duration_seconds is required".to_string())
+        })?)
+        .send()
+        .await
+        .map_err(Error::from)
+}
+
+fn generate_rds_iam_token(
+    db_hostname: &str,
+    region: Region,
+    port: u16,
+    db_username: &str,
+    credentials: &Credentials,
+    timestamp: SystemTime,
+) -> std::result::Result<String, SigningError> {
+    let signer = signer::SigV4Signer::new();
+    let mut operation_config = OperationSigningConfig::default_config();
+    operation_config.signature_type = HttpSignatureType::HttpRequestQueryParams;
+    operation_config.expires_in = Some(Duration::from_secs(15 * 60));
+    let request_config = RequestConfig {
+        request_ts: timestamp,
+        region: &SigningRegion::from(region),
+        service: &SigningService::from_static("rds-db"),
+        payload_override: None,
+    };
+    let mut request = http::Request::builder()
+        .uri(format!(
+            "http://{db_hostname}:{port}/?Action=connect&DBUser={db_user}",
+            db_hostname = db_hostname,
+            port = port,
+            db_user = db_username
+        ))
+        .body(SdkBody::empty())
+        .expect("valid request");
+    let _signature = signer.sign(
+        &operation_config,
+        &request_config,
+        credentials,
+        &mut request,
+    )?;
+    let mut uri = request.uri().to_string();
+    assert!(uri.starts_with("http://"));
```

Comment:
does this not support/expect `https://` since it's talking to the aws metadata service so it's expected to _always_ be private to the vpc the host is running on?

Reply:
This function is copied from the library example and since it works, I haven't changed it at all.

https://docs.rs/aws-sig-auth/0.54.1/aws_sig_auth/#generate-rds-iam-token